bokeh/__init__.py

```diff
@@ -1,285 +1,29 @@
-        r = requests.post(issues_url,
-                          auth=(ghuser, ghpass),
-                          headers={'Content-Type': 'application/json'},
-                          data=json.dumps(data))
-        if r.status_code == 201:
-            g = requests.get(issues_url)
-            if number is None:
-                print("Issue successfully submitted.")
-                if browser:
-                    webbrowser.open_new(g.json()[0].get("html_url"))
-            else:
-                print("Comment successfully submitted.")
-                g = requests.get(issues_url)
-                if browser:
-                    webbrowser.open_new(g.json()[-1].get("html_url"))
-        else:
-            print("Something failed, please check your username and password.")
-    else:
-        print("Issue not submitted.")
-
-def test(verbosity=1, xunitfile=None, exit=False):
-    """ Run the full Bokeh test suite, and output the results of the tests
-    to sys.stdout.
-
-    This function uses nosetests to discover which tests to run, and will
-    run tests in any 'tests' subdirectory within the Bokeh module.
-
-    Args:
-        verbosity (int, optional) :
-            Acceptatable values are 0 (less verbose) to 2 (most verbose)
-
-        xunitfile (str, optional) :
-            Write xunit-style XML test results to a given filename. This
-            is useful for running tests on a CI server. (default: None)
-
-        exit (bool, optional) :
-            Whether to return or exit. If True, call sys.exit with an
-            error code after the tests are finished. (default: False)
-
-    Returns:
-        int : nose return code
-
-    """
-    import nose, os, sys
-
-    argv = ['nosetests', '--verbosity=%d' % verbosity]
-
-    # Output an xunit file if requested
-    if xunitfile:
-        argv.extend(['--with-xunit', '--xunit-file=%s' % xunitfile])
+"""
+from __future__ import absolute_import, print_function
 
-    # Set the logging level to warn
-    argv.extend(['--logging-level=WARN'])
+# configure Bokeh version
+from .util.version import __version__; __version__
 
-    # Add all 'tests' subdirectories to the options
-    rootdir = os.path.dirname(__file__)
-    for root, dirs, files in os.walk(rootdir):
-        if 'tests' in dirs:
-            testsdir = os.path.join(root, 'tests')
-            argv.append(testsdir)
-            print('Test dir: %s' % testsdir[len(rootdir)+1:])
+# configure Bokeh logger
+from .util import logconfig
+del logconfig
 
-    # print versions (handy when reporting problems)
-    print_versions()
-    sys.stdout.flush()
+# imports below are names we want to make available in the bokeh
+# module as transitive imports
 
-    # Run the tests
-    return nose.main(argv=argv, exit=exit)
+from . import sampledata; sampledata
+from .settings import settings; settings
+from .util.testing import print_versions; print_versions
+from .util.testing import report_issue; report_issue
```

Comment:
And @damianavila as nice an idea as this is I don't think anyone has used it and I am not sure how to encourage it more. It might do to remove it just to have less to maintain. (just an idea to consider)